sql/tsql/TSqlLexer.g4

```diff
@@ -551,6 +369,7 @@ GB:                                    'GB';
 GETDATE:                               'GETDATE';
 GETUTCDATE:                            'GETUTCDATE';
 GLOBAL:                                'GLOBAL';
+GO_BATCH:                              [\r\n]* [\t ]* 'GO'[\t\n\r ];
```

Comment:
I don't think such construction is optimal from the performance perspective because it conflicts with `SPACE` token. Can it be replaced with post predicate like this?

```suggestion
GO_BATCH:                              'GO' {isNewLine(-2)}? [\t\n\r ];
```

We usethe same approach for [plsql](https://github.com/antlr/grammars-v4/blob/master/sql/plsql/PlSqlLexer.g4#L2345) and [php](https://github.com/antlr/grammars-v4/blob/master/php/PhpLexer.g4#L46) grammars.